docs/source/guides/schema-design.md

```diff
@@ -0,0 +1,198 @@
+---
+title: Schema design
+---
+
+GraphQL schemas are at their best when they are designed around the needs of client applications.  When a team is building their first GraphQL schema, they might be tempted to create literal mappings on top of existing database collections or tables using CRUD-like root fields but it’s important to consider how this could be disadvantageous.
+
+While this literal database-to-schema mapping may be a fast way to get up and running, we strongly suggest avoiding it and instead building the schema around based on how the GraphQL API will be used by the front-end.
+
+## Style conventions
+
+The GraphQL specification is flexible in the style that it dictates and doesn't impose specific naming guidelines. In order to facilitate development and continuity across GraphQL deployments, we suggest the following style conventions:
+
+* **Fields**: are recommended to be written in `camelCase`, since the majority of consumers will be client applications written in JavaScript.
+* **Types**: should be `PascalCase`.
+* **Enums**: should have their name in `PascalCase` and their values in `ALL_CAPS` to denote their special meaning.
+
+## Wrapping documents with gql
+
+There are two common ways to write GraphQL schemas and queries. The first is to write queries into a `.graphql` file and import them into your other files for usage. The second is to write them wrapped them with the `gql` tag provided by the `graphql-tag` library.
```

Comment:
The most common one right now is by far to just have bare strings, so this paragraph seems inaccurate.

Reply:
Agree this should be changed.  We shouldn't pretend like the most popular use case doesn't exist, but we have an increasing number of reasons to recommend using `gql`  to enable syntax-highlighting and to allow for future functionality.  It's now exported/available from `apollo-server` itself so this seems like an opportunity to mention that bare-strings are an option, but that we recommend using `gql` because of _x_, _y_ and _z_.